### PR TITLE
Feature/sw 5 02/112 design  flow backend

### PR DESCRIPTION
**Pull request by JacobInberg**
# Description

Refactores a lot of code and renames a lot of code to actually follow conventions.

Please look everything through and ensure that if anything needs to be changed it is said

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes #\<issue_no>

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration*

**Development Configuration**
*Type "flutter --version" and "dart --version" in your CMD to check versions.*

* Flutter version:
* Dart version:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [x] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: FoodplannerApi/Controller/UsersController.cs [hunk 1]
```diff
@@ -35,102 +37,191 @@ public async Task<IActionResult> GetBearerTest()
...
+    [Authorize(Roles = "Child, Parent")]
+    public async Task<IActionResult> Get(int id)
+    {
+        var users = await _userService.GetUserByIdAsync(id);
+        if (users == null)
+        {
+            return NotFound();
+        }
+        return Ok(users);
+    }
```
Comment (JacobStokholms) [inline FoodplannerApi/Controller/UsersController.cs:174]: admin?

File: FoodplannerDataAccessSql/Account/UserRepository.cs [hunk 2]
```diff
@@ -214,6 +214,46 @@ public async Task<bool> UpdateRoleApprovedAsync(int id, bool roleApproved)
...
+                    LastName = userUpdateDTO.LastName,
+                    Email = userUpdateDTO.Email,
+                });
+                return result;
+            }
+        }
+
+        public async Task<string> UpdatePasswordAsync(string password, int id)
+        {
+            var sql = "UPDATE users SET password = @Password WHERE id = @Id RETURNING password";
```
Comment (JacobStokholms) [inline FoodplannerDataAccessSql/Account/UserRepository.cs:248]: Return password, it is encrypted yes ?

**Review by JacobStokholms — COMMENTED** [review]
Fix the comment then good :)

**Review by JacobStokholms — APPROVED** [review]

**Comment by JacobInberg** [comment]
PR til endelige prototype

**Pull request merged**